### ISSUE TITLE
Improvement: Add to the feature the capability of handling an array of styles

### ISSUE THREAD

**Issue by gmvtrmaga** [TDesjardins/gwt-ol3#122]
As OL3 development team explained in this [issue](https://github.com/openlayers/openlayers/issues/6597), the feature is able to handle several styles (in the past, it could only manage front and back style).

I've implemented the following code in the gwt library in order to test this:
`public native void setStyleArray(Feature feature, Style[] styleArray)/*-{
		feature.setStyle(styleArray);
	}-*/; `
and it worked perfectly, but this change broke the feature style getter.

In conclusion, Feature class should be modified if we want it to manage arrays of Styles.

Regards, and thanks in advance.

**Comment by TDesjardins** [comment]
You could already do this with `Featue:setStyle(GenericFunction<Double, Style[]> styleFunction)` but yes the simple style array setter should also be supported. This can be done by simply overloading it... and yes we need a solution for the UGLY OpenLayers multitype getters 😠 .

**Comment by gmvtrmaga** [comment]
What would be your approach regarding the multiple getter selection? Have you got any suggestions?

**Comment by TDesjardins** [comment]
Yes, there is an idea I am working on. I have to test it for general use. The solution has to avoid 
inconsistencies because it isn't the job of the API user to care about that. 😉

**Comment by TDesjardins** [comment]
@gmvtrmaga Could you test this? I think there also a lot of other places where we have to handle "multi getters".

**Comment by gmvtrmaga** [comment]
I have just tried it and it works like a charm :)

Regarding the multi getters, it seems so, but I haven't found it yet. I will be checking this during the next days.

Thank you for your fast implementation.

Regards.
